### PR TITLE
Update for `config` being peer of `src`.

### PR DESCRIPTION
**Pull request by rwjblue**
Implements changes needed to support https://github.com/glimmerjs/glimmer-application-pipeline/pull/73.

### CODE REVIEW

File: files/src/config/module-map.ts [hunk 1]
```diff
@@ -1,6 +0,0 @@
-/**
```
Comment (rwjblue) [inline files/src/config/module-map.ts]: We may want to keep this around, but move it up into the root.

Without it, I believe that VSCode and other TS aware editors will show a warning that this module cannot be found...

Thoughts @dgeb / @mmun  / @tomdale?

File: files/package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
   },
   "devDependencies": {
     "@glimmer/application": "^0.4.0",
-    "@glimmer/application-pipeline": "^0.5.2",
+    "@glimmer/application-pipeline": "^6.0.0",
```
Comment (dgeb) [inline files/package.json]: I hope you mean `0.6.0`

Comment (rwjblue) [inline files/package.json]: lololol, yes (thank you!)

File: files/src/config/module-map.ts (hunk 1 above)
Comment (rwjblue) [inline files/src/config/module-map.ts]: This is somewhat moot now, I brought this back as a *.d.ts only. This prevents editors from warning about this module missing, but also makes it clear that its only a definition file...

Comment (dgeb) [inline files/src/config/module-map.ts]: Ok, `.d.ts` is ideal here 👍

File: files/config/module-map.d.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/**
+ * This is just a placeholder file to keep TypeScript aware editors happy. At build time,
+ * it will be replaced with a complete map of resolvable module paths => rolled up contents.
+ */
+
+export interface Dict<T> {
+  [index: string]: T;
+}
```
Comment (dgeb) [inline files/config/module-map.d.ts]: I think we can just `import Dict from '@glimmer/di';` instead of redefining

Comment (rwjblue) [inline files/config/module-map.d.ts]: Hmm, that makes sense but we don't actually depend on @glimmer/di...

**Comment by rwjblue** [comment]
This blocked on https://github.com/glimmerjs/glimmer-blueprint/pull/31, which will make adding a helpful warning/error message **and** upgrading from one version to another much easier.

**Pull request merged**